src/librbd/AioObjectRequest.cc

```diff
@@ -623,6 +624,13 @@ void AioObjectTruncate::send_write() {
   }
 }
 
+void AioObjectTrim::pre_object_map_update(uint8_t *new_state) {
+  *new_state = OBJECT_PENDING;
+
+  // object map state (before pre) decides if a post updation is needed
+  need_post = m_ictx->object_map->update_required(m_object_no, *new_state);
```

Comment:
Why is this needed? Shouldn't https://github.com/vshankar/ceph/blob/master/src/librbd/AioObjectRequest.cc#L485 cover the cause when the PRE didn't update the object map (since https://github.com/vshankar/ceph/blob/master/src/librbd/ObjectMap.cc#L98 would essentially no-op the update if the object is already non-existent), thus the POST would be no-op as well (since it's still flagged as non-existent).